pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -61,6 +77,72 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 	return nil
 }
 
+// ensureExternalClusterConfig ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalClusterConfig(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: secretNamespace,
+		},
+		Data: map[string][]byte{
```

Comment:
I don't think you need this Data section, just get it from what is going to be fetched.

Reply:
Sure, will remove that the Data section, Since we are already catching it at line 106 I believe.